mobly/controllers/android_device.py

```diff
@@ -812,12 +826,16 @@ def reboot(self):
         if self.is_bootloader:
             self.fastboot.reboot()
             return
+        snippet_info = self._get_active_snippet_info()
```

Comment:
Just curious why you have to capture these? Isn't it enough to iterate through _snippet_clients after reboot and reconnect them?